FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp

```diff
@@ -101,28 +101,44 @@ DEF_OP(LoadNamedVectorConstant) {
     }
   }
   // Load the pointer.
-  ldr(TMP1, STATE_PTR(CpuStateFrame, Pointers.Common.NamedVectorConstantPointers[Op->Constant]));
+  auto GenerateMemOperand = [this](uint8_t OpSize, uint32_t NamedConstant, FEXCore::ARMEmitter::Register Base) {
+    const auto ConstantOffset = offsetof(FEXCore::Core::CpuStateFrame, Pointers.Common.NamedVectorConstants[NamedConstant]);
 
+    if (ConstantOffset <= 255) {
+      return ARMEmitter::ExtendedMemOperand(Base.X(), ARMEmitter::IndexType::OFFSET, ConstantOffset);
+    }
+    else if ((ConstantOffset & (OpSize - 1)) == 0 &&  ConstantOffset / OpSize <= 4095 ) {
+      return ARMEmitter::ExtendedMemOperand(Base.X(), ARMEmitter::IndexType::OFFSET, ConstantOffset);
```

Comment:
copypaste fail?

Reply:
Just a bit tricky here.
The ldr/str code emitters that consume the returned `ARMEmitter::ExtendedMemOperand` support selecting either the 9-bit signed unscaled instruction or 12-bit unsigned scaled instruction depending on constant offset.
I just derped and had if checks to return the same thing.

Combined the if statements so it doesn't look funny when reading it.